### PR TITLE
compilation and submodules

### LINKED ISSUE #267: Add unit tests for `BaseBundler`
```[tasklist]
### Tasks
- [ ] `multicall` should set the correct the initiator
- [ ] `_revert` bubbles up well error messages
```

### LINKED ISSUE #397: Fix issue with URD v1.0.0 solc version

### LINKED ISSUE #405: Check for functions signatures clash
with morpho's functions

### LINKED ISSUE #410: harmonise compilation of libs
This is a solution to: "how to tell forge to compile these contracts because we need its bytecode ready to be deployed in tests"

but it is another kind of solution than the one we went for on this repository: enforce `yarn build:metamorpho` for example when running tests

Your solution is arguably better because it is enforced at forge's level, not at the command level
But then ideally we also do this for MetaMorpho, IRM, Oracle! In another PR

_Originally posted by @Rubilmax in https://github.com/morpho-org/morpho-blue-bundlers/pull/403#discussion_r1500308381_

### LINKED ISSUE #411: submodules point to morpho-labs org

### PR DESCRIPTION
**Pull request by MathisGD**
Fixes #411 
Fixes #397 
Fixes #410 
Fixes #267
Fixes #405 
and updates morpho submodules

### CODE REVIEW

File: test/forge/helpers/BaseTest.sol [hunk 1]
```diff
@@ -64,7 +64,7 @@ abstract contract BaseTest is Test {
     bytes[] internal callbackBundle;
 
     function setUp() public virtual {
-        morpho = IMorpho(_deploy("lib/morpho-blue/out/Morpho.sol/Morpho.json", abi.encode(OWNER)));
+        morpho = IMorpho(_deploy("out/Morpho.sol/Morpho.json", abi.encode(OWNER)));
```
Comment (Rubilmax) [inline test/forge/helpers/BaseTest.sol:67]:
```suggestion
        morpho = IMorpho(_deploy("Morpho", abi.encode(OWNER)));
```

You may already know it but this should be enough and doesn't depend on forge's output folder ^^

**Review by Rubilmax — DISMISSED** [review]

File: test/forge/helpers/MetaMorphoLocalTest.sol [hunk 2]
```diff
@@ -29,7 +29,7 @@ abstract contract MetaMorphoLocalTest is LocalTest {
 
         vault = IMetaMorpho(
             _deploy(
-                "lib/metamorpho/out/MetaMorpho.sol/MetaMorpho.json",
+                "out/MetaMorpho.sol/MetaMorpho.json",
```
Comment (MathisGD) [inline test/forge/helpers/MetaMorphoLocalTest.sol:32]:
```suggestion
                "MetaMorpho",
```

File: src/mocks/AdaptiveCurveIrmMock.sol
[the remark this answers is not in the document]
Comment (MathisGD) [reply src/mocks/AdaptiveCurveIrmMock.sol:1]: nice catch

File: src/mocks/AdaptiveCurveIrmImport.sol [hunk 3]
```diff
@@ -0,0 +1,6 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.0;
```
Comment (Rubilmax) [inline src/mocks/AdaptiveCurveIrmImport.sol:2]: No strict pragma here which is inconsistent with MetaMorpho and Morpho but not a big deal

**Review by Rubilmax — DISMISSED** [review]

File: src/mocks/AdaptiveCurveIrmImport.sol (hunk 3 above)
Comment (MathisGD) [reply src/mocks/AdaptiveCurveIrmImport.sol:2]:
```suggestion
pragma solidity 0.8.19;
```

File: lib/morpho-blue-irm
Comment (MerlinEgalite) [inline lib/morpho-blue-irm:1]: note that the irm code is updated at this commit

File: lib/morpho-utils
Comment (MerlinEgalite) [inline lib/morpho-utils:1]: same

File: lib/universal-rewards-distributor
Comment (MerlinEgalite) [inline lib/universal-rewards-distributor:1]: same

File: test/forge/SelectorClashTest.sol [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import "../../lib/morpho-blue/src/interfaces/IMorpho.sol";
+import "../../lib/universal-rewards-distributor/src/interfaces/IUniversalRewardsDistributor.sol";
+import "../../lib/openzeppelin-contracts/contracts/token/ERC20/extensions/IERC20Permit.sol";
+import "../../lib/openzeppelin-contracts/contracts/interfaces/IERC4626.sol";
+import "../../lib/openzeppelin-contracts/contracts/token/ERC20/extensions/ERC20Wrapper.sol";
+
+// The bundler can do call to arbitrary contracts, we make sure no selectors clash by inheriting all the interfaces in
+// one single contract.
+abstract contract SelectorClashTest is IMorpho, IUniversalRewardsDistributor, IERC4626, IERC20Permit, ERC20Wrapper {
+    // Overriedes of functions that are present in multiple interfaces contract ("acknoledged clashes").
```
Comment (MerlinEgalite) [inline test/forge/SelectorClashTest.sol:13]:
```suggestion
    // Overrides of functions that are present in multiple interfaces contract ("acknowledged clashes").
```

**Review by MerlinEgalite — APPROVED** [review]

**Review by Rubilmax — APPROVED** [review]

File: test/forge/SelectorClashTest.sol
Comment (QGarchery) [inline test/forge/SelectorClashTest.sol:1]: Nice ! Do we want to also add `deposit` and `withdraw` of WNative ?

File: test/forge/BaseBundlerLocalTest.sol [hunk 5]
```diff
@@ -25,3 +25,27 @@ contract BaseBundlerLocalTest is LocalTest {
         bundler.multicall(bundle);
     }
 }
+
+contract BaseBundlerLocalEnshrinedTest is BaseBundler, LocalTest {
```
Comment (QGarchery) [inline test/forge/BaseBundlerLocalTest.sol:29]: We could move that to its own file

**Review by QGarchery — DISMISSED** [review]

File: test/forge/BaseBundlerLocalTest.sol (hunk 5 above)
Comment (Rubilmax) [reply test/forge/BaseBundlerLocalTest.sol:29]: Ideally named `BaseBundlerMulticallLocalTest`?

Comment (Rubilmax) [reply test/forge/BaseBundlerLocalTest.sol:29]: Why is it separated?

File: test/forge/SelectorClashTest.sol
Comment (MathisGD) [reply test/forge/SelectorClashTest.sol:1]: nice catch, done

**Review by Rubilmax — DISMISSED** [review]

File: test/forge/BaseBundlerEnshrinedLocalTest.sol [hunk 6]
```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.0;
+
+import {BaseBundler} from "src/mocks/bundlers/BaseBundlerMock.sol";
+
+import "./helpers/LocalTest.sol";
+
+contract BaseBundlerLocalEnshrinedTest is BaseBundler, LocalTest {
```
Comment (QGarchery) [inline test/forge/BaseBundlerEnshrinedLocalTest.sol:8]:
```suggestion
contract BaseBundlerEnshrinedLocalTest is BaseBundler, LocalTest {
```
to match the file name

**Review by QGarchery — APPROVED** [review]

**Review by QGarchery — APPROVED** [review]

**Review by QGarchery — APPROVED** [review]

**Review by MerlinEgalite — APPROVED** [review]

**Pull request merged**